acceptance/tests/pxp-module-puppet/run_puppet_syntax_chars.rb

```diff
@@ -0,0 +1,61 @@
+require 'pxp-agent/test_helper.rb'
+require 'puppet/acceptance/environment_utils'
+
+test_name 'C98237 - Run puppet with syntax significant characters in Puppet code' do
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  env_name = test_file_name = File.basename(__FILE__, '.*')
+  environment_name = mk_tmp_environment(env_name)
+
+  step 'On master, create a new environment that includes syntax significant characters in resources and will result in Changed' do
+    site_manifest = "#{environmentpath}/#{environment_name}/manifests/site.pp"
+    create_remote_file(master, site_manifest, <<-SITEPP)
+node default {
+  notify {'\\'\\"&+/\\()?,.#!@$_-~':}
```

Comment:
I think you're right @MikaelSmith - this shouldn't be an acceptance test unless we suspect different behaviours on different platforms. If we trust the Ruby execution of pxp-module-puppet to be consistent across all agent platforms; this can be a spec test.